xarray/coding/times.py

```diff
@@ -542,21 +556,25 @@ def _should_cftime_be_used(source, target_calendar, use_cftime):
         if _is_standard_calendar(target_calendar):
             if _is_numpy_compatible_time_range(source):
                 # Conversion is possible with pandas, force False if it was None
-                use_cftime = False
+                use_cftime_out = False
             elif use_cftime is False:
                 raise ValueError(
                     "Source time range is not valid for numpy datetimes. Try using `use_cftime=True`."
                 )
+            else:
+                use_cftime_out = True
```

Comment:
Here it was possible to return None previously. From reading the docstring and the name of the function I don't think it was intended to return None here, so I just followed the upper if else format and return True in this case.

Reply:
I always find it easier to use early returns in such nested if functions, but I guess that's a personal preference.

Reply:
Thank you, the new version does indeed look better.